### ISSUE TITLE
Docker issue (with port)

### ISSUE THREAD

**Issue by johnrbeaumont** [davidADSP/Generative_Deep_Learning_2nd_Edition#6]
Hi --

I am trying to build the docker file on a Mac M1. I think I've done everything correctly, but I get the following error when I build the docker file - any ideas:

```
WARN[0000] The "JUPYTER_PORT" variable is not set. Defaulting to a blank string. 
WARN[0000] The "JUPYTER_PORT" variable is not set. Defaulting to a blank string. 
WARN[0000] The "TENSORBOARD_PORT" variable is not set. Defaulting to a blank string. 
WARN[0000] The "TENSORBOARD_PORT" variable is not set. Defaulting to a blank string. 
WARN[0000] The "JUPYTER_PORT" variable is not set. Defaulting to a blank string. 
services.app.ports array items[0,1] must be unique
```

**Comment by karstenjMSFT** [comment]
yep - you have to set the .env and can use the sample.env in the project

**Comment by FabioLddo** [comment]
Would you be so kind to explain what do you mean by setting the .env file?
I have already put all the possible paths to the sample.env file in the .yml but docker It does not find the variables.
Also I've already put kaggle name and key but still get this:

time="2024-02-06T21:38:15+01:00" level=warning msg="The \"JUPYTER_PORT\" variable is not set. Defaulting to a blank string."
time="2024-02-06T21:38:15+01:00" level=warning msg="The \"JUPYTER_PORT\" variable is not set. Defaulting to a blank string."
time="2024-02-06T21:38:15+01:00" level=warning msg="The \"TENSORBOARD_PORT\" variable is not set. Defaulting to a blank string."
time="2024-02-06T21:38:15+01:00" level=warning msg="The \"TENSORBOARD_PORT\" variable is not set. Defaulting to a blank string."
time="2024-02-06T21:38:15+01:00" level=warning msg="The \"JUPYTER_PORT\" variable is not set. Defaulting to a blank string."
no port specified: :<empty>
